internal/lockclient/simple_client.go

```diff
@@ -30,21 +33,34 @@ func NewSimpleClient(config lockservice.SimpleConfig, cache cache.LRUCache) *Sim
 
 var _ Client = (*SimpleClient)(nil)
 
-// Acquire makes a HTTP call to the lockserver and acquires the lock.
-// The errors invloved may be due the HTTP errors or the lockservice errors.
+// Acquire allows the user process to acquire a lock.
 func (sc *SimpleClient) Acquire(d lockservice.Descriptors) error {
-	endPoint := sc.config.IP() + ":" + sc.config.Port() + "/acquire"
-
-	isInCache := sc.cache.GetElement(cache.NewSimpleKey(d.ID()))
+	return sc.acquire(d)
+}
 
-	if isInCache == nil {
-		return lockservice.ErrFileAcquired
+// acquire makes an HTTP call to the lockserver and acquires the lock.
+// This function makes the acquire call and doesn't care about the contention
+// on the lock service.
+// The errors involved may be due the HTTP errors or the lockservice errors.
+//
+// Currently acquire doesn't order the user processes that request for the lock.
+func (sc *SimpleClient) acquire(d lockservice.Descriptors) error {
+	// Check for existance of a cache and check
+	// if the element is in the cache.
+	if sc.cache != nil {
+		_, err := sc.getFromCache(lockservice.ObjectDescriptor{ObjectID: d.ID()})
+		// Since there can be network errors, we have this double check.
+		if err != nil && err != lockservice.ErrCheckacquireFailure {
```

Comment:
I've requested a change within the implementation of the `getFromCache()` function. The corresponding change would have to be made here.

Reply:
Im not sure what you mean. Please explain :D

Reply:
see the comment above :P